### DISCUSSION TITLE
疑问：“今天吃什么”会触发吃派蒙吗

### DISCUSSION THREAD

**Discussion by mark9804** [Arondight/Adachi-BOT#243] [Q&A] [answered]
https://github.com/Arondight/Adachi-BOT/blob/master/src/plugins/tools/menu.js#L10-L13

在这个写法当中，`breakfast`，`lunch`，`dinner`感觉好像不可能出现false的情况，所以三目运算理论上是不是只会从菜单里挑选，不会出现条件为false然后跳转到`favFood`的情况？

我是一直没试出来，就把判断条件改成了`Math.random() < 0.9`，不知道[这个改动](https://github.com/Mark9804/Adachi-BOT/commit/a6bb03d0b33e928dfce2a4655a8175242968d193#diff-a4d81fcf56044f93389da083efb40717ff93a9457cd11c919be08302ba4a5676R23)是不是多余的

**Comment by Arondight** [comment, marked as answer]
配置文件是个随便改的东西，这三个值都可能是 undefined ，只要把配置文件清空就行了

**Reply by mark9804** [reply]
这样，原来派蒙是用来兜底的真应急食品啊……

**Reply by Arondight** [reply]
> 
> 
> 这样，原来派蒙是用来兜底的真应急食品啊……

是这样的，兜底的，这么一说似乎把 `breakfast ? breakfast[getRandomInt(breakfast.length) - 1] : favFood` 中的真值判断改成 `Array.isArray(breakfast)` 比较好，毕竟当前这么写派蒙还是有可能兜不住的，例如配置文件中写成 `breakfast: 苹果` 就兜不住了。

---

类似的还有很多兜底的代码，就像这种。

https://github.com/Arondight/Adachi-BOT/blob/7d1e29eaf330b9cf3201f5344bd44fbcfdbd4677/src/utils/config.js#L436

实际上这种情况只有在非常极端的情况下才会发生，因为没有任何插件才会走到这一步，这时候只有主程序自行调用这个数据结构才会走到这里，这几乎不可能而且毫无意义。

---

也有极端条件下会发生但确实有意义的例如这种。

https://github.com/Arondight/Adachi-BOT/blob/7d1e29eaf330b9cf3201f5344bd44fbcfdbd4677/src/plugins/music/music.js#L111-L112

我一直很头疼的就是这堆代码历史遗留问题太多，很多地方本来应该有兜底但是没有，我也不可能一行行地去加，只能什么时候有人点了炒饭再去处理了……
